src/frontend/src/tests/lib/stores/transactions.store.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { ETHEREUM_TOKEN_ID, ICP_TOKEN_ID } from '$env/tokens.env';
+import type { IcTransactionUi } from '$icp/types/ic';
+import { nowInBigIntNanoSeconds } from '$icp/utils/date.utils';
+import { initTransactionsStore } from '$lib/stores/transactions.store';
+
+describe('transactions.store', () => {
+	const createMockTransaction = (id: string): IcTransactionUi => ({
+		id,
+		timestamp: nowInBigIntNanoSeconds(),
+		type: 'send',
+		value: BigInt(100),
+		from: 'sender',
+		to: 'receiver',
+		status: 'pending'
+	});
+
+	const createCertifiedTransaction = (id: string) => ({
+		certified: true,
+		data: createMockTransaction(id)
+	});
+
+	const tokenId = ICP_TOKEN_ID;
+
+	describe('prepend', () => {
+		it('should add transactions at the beginning of the list', () =>
+			new Promise<void>((done) => {
+				const store = initTransactionsStore<IcTransactionUi>();
+
+				const initialTx = [createCertifiedTransaction('tx1')];
+				store.prepend({ tokenId, transactions: initialTx });
+
+				const newTx = [createCertifiedTransaction('tx2')];
+				store.prepend({ tokenId, transactions: newTx });
+
+				store.subscribe((state) => {
+					expect(state?.[tokenId]).toHaveLength(2);
+					expect(state?.[tokenId]?.[0].data.id).toBe('tx2');
+					expect(state?.[tokenId]?.[1].data.id).toBe('tx1');
+
+					done();
+				})();
+			}));
+
+		it('should deduplicate transactions with same id', () =>
+			new Promise<void>((done) => {
+				const store = initTransactionsStore<IcTransactionUi>();
+
+				const tx = createCertifiedTransaction('tx1');
+				store.prepend({ tokenId, transactions: [tx] });
+
+				const newTx = [createCertifiedTransaction('tx2')];
+				store.prepend({ tokenId, transactions: newTx });
+
+				store.prepend({ tokenId, transactions: [tx] });
+
+				store.subscribe((state) => {
+					expect(state?.[tokenId]).toHaveLength(2);
+
+					done();
+				})();
+			}));
+	});
+
+	describe('append', () => {
```

Comment:
maybe here the same test to check if no duplicates? as the one for prepend?

Reply:
that's actually a good point, turns out `append` do not deduplicate. Should we improve the store in another PR and provide the test at the same time?

Reply:
Yes! I didn't even know that it wasn't doing it, good thing you tested!